airflow/configuration.py

```diff
@@ -580,10 +589,24 @@ def getint(self, section, key):
     def getfloat(self, section, key):
         return float(self.get(section, key))
 
+    def remove_option(self, section, option):
+        try:
+            ConfigParser.remove_option(self, section, option)
+        except ConfigParser.NoSectionError:
+            pass
+        try:
+            ConfigParser.remove_option(self.defaults, section, option)
+        except ConfigParser.NoSectionError:
+            pass
+
     def read(self, filenames):
         ConfigParser.read(self, filenames)
         self._validate()
 
+    def has_option(self, section, option):
```

Comment:
I don't think that we want to override `has_option`. Allowing `has_option` to cascade to `self.defaults` muddies the difference between set options and default options.

Alternatively, if we want to adopt the semantics that `has_option` will return `True` if we are able to retrieve the option, we would want to include all possible means for retrieving an option (env var, cmd, defaults, etc.)

Probably, best to just not override `has_option` and leave it meaning that the config _explicitly_ has the option.